src/components/ControlBar/ControlBar.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import React from 'react'
+import { View, ActivityIndicator, } from 'react-native'
+
+import { Icon, TouchableButton, } from '../index'
+import styles from './style'
+
+type Props = {
+  flashMode: boolean,
+  isCameraReady: boolean ,
+  isImage: boolean,
+  toggleType: () => void,  
+  takePicture: () => void,
+  toggleFlashMode: () => void,
+}
+
+const ControlBar = ({ 
+  flashMode, 
+  takePicture, 
+  toggleFlashMode, 
+  isCameraReady, 
+  isImage, 
+  toggleType, 
+}: Props) => (
+  <View style={styles.container}>
+    <TouchableButton style={styles.btnIcons} onPress={toggleFlashMode}>
+      <Icon name='bolt' color={flashMode ? '#3EE7AD' : '#e6e5e6'}/>
+    </TouchableButton>
+    <TouchableButton style={styles.cameraBtn} onPress={isCameraReady ? takePicture : null}>
```

Comment:
Maybe it's better just to disable button if a camera isn't ready?

Reply:
fixed